### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by lshig**
**Description of changes**

- yarn.lock and package.json changed to add dedent and rehype for the test files
- Added Jest tests
- Added Jest snapshots

**Reviewers**

@davidtheclark

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+describe('rehypeHighlightCodeBlock', () => {
+  test('mocks syntax highlighting properly', () => {
+    const result = processHtml(dedent`
+      <div>
+        <p>foo</p>
+        <pre><code>p { color: red }</code></pre>
+        <p>bar</p>
+      </div>
+    `);
+    expect(result).toMatchSnapshot();
```
Comment (davidtheclark) [inline test.js]: In addition to these snapshot checks, we should also check and make sure that the spy function (created with `jest.fn()`) was called with the correct arguments.

Comment (lshig) [reply test.js]: @davidtheclark I added the updated tests in my new commit (	e788691) and made a small tweak in the regex in index.js. Let me know if I missed anything or if you have any comments/questions.

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  test('mocks syntax highlighting with language-* class', () => {
+    const result = processHtml(dedent`
+      <div>
+        <p>foo</p>
+        <pre><code class="language-css">p { color: red }</code></pre>
+        <p>bar</p>
+      </div>
+    `);
+    expect(mockHighLightSpy.mock.calls[0].length).toBe(2);
+    expect(mockHighLightSpy.mock.calls[0][1]).toBe('css');
```
Comment (davidtheclark) [inline test/test.js]: I think we should be able to make an assertion about both arguments. This line might be able to replace 32-33:

```js
expect(mockHighlightSpy).toHaveBeenCalledWith(
  `p { color: red }`,
  'css'
);
```

We need to assert that the correct first argument, the content of the code block, is passed in.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    `);
+    expect(mockHighLightSpy.mock.calls.length).toBe(0);
+    expect(result).toMatchSnapshot();
+  });
+
+  test('does nothing to code block, missing pre > code', () => {
+    const result = processHtml(dedent`
+      <div>p { color: red }</div>
+    `);
+    expect(mockHighLightSpy.mock.calls.length).toBe(0);
```
Comment (davidtheclark) [inline test/test.js]: This is fine. But FYI there's also a shorthand method for this: `expect(mockHighLightSpy).toHaveBeenCalledTimes(0)`.

**Review by davidtheclark — COMMENTED** [review]
💯  One more suggestion for you.

File: test/test.js (hunk 2 above)
Comment (lshig) [reply test/test.js]: Nice! That's easier to read.

File: test/test.js (hunk 3 above)
Comment (lshig) [reply test/test.js]: Thanks again! I added these quick changes in to the new commit (a32a68d).

**Review by davidtheclark — APPROVED** [review]
🍨

**Pull request merged**